app.js

```diff
@@ -0,0 +1,146 @@
+//基礎編 問題
+//Q1 変数
+let nickname  = 'ごっしー';
+let age = (28);
+let greet = '私のニックネームは' + nickname + 'です。' + '年齢は' + age + '歳です。';
+console.log(greet);
+
+//Q2 配列
+let languages = ['JavaScript', 'PHP', 'Ruby', 'Python','Go'];
+let text ='私の好きな言語は' + languages[0] + 'です。'+ '次は' + languages[3] + '勉強してみたいです。';
+console.log('languages[0] => ', languages[0]);
+console.log('languages[1] => ', languages[1]);
+console.log('languages[2] => ', languages[2]);
+console.log('languages[3] => ', languages[3]);
+console.log('languages[4] => ', languages[4]);
+console.log(text)
+
+//Q3 オブジェクト
+let user = {
+  name: 'John',
+  age: 26,
+  bloodType: 'A',
+  favorite: 'card',
+};
+console.log(user.age)
+
+//Q4 配列 × オブジェクト
+let playerList = [
+  {
+    name: 'John',
+    age: 26,
+    favorites: ['Card Game', 'Basket Ball', 'Programming'],
+  },
+  {
+    name: 'Bob',
+    age: 33,
+    favorites: ['Tinder', 'The Legend of Zelda'],
+  },
+  {
+    name: 'Michael',
+    age: 22,
+    favorites: ['Football', 'Smash Bros.'],
+  },
+];
+console.log(playerList[1].favorites[1])
+
+//Q5 四則演算
+let avgage = ((playerList[0].age + playerList[1].age + playerList[2].age) /3)
+console.log(avgage)
+
+//Q6 関数
+function sayHello (){
+  console.log('Hello');
+}
+sayHello();
+
+let sayWorld = function (){
+  console.log('World');
+}
+sayWorld();
+
+//Q7 メソッド
+user.birthday ='2000-09-27';
+
+user.sayHello = function(){
+  console.log('Hello！');
+}
+user.sayHello();
+
+//Q8 引数
+let calc = {};
+
+calc.add = function(x, y){
+  console.log(x + y);
+}
+calc.add(3, 4);
+
+calc.subtract = function(x, y){
+  console.log(x - y);
+}
+calc.subtract(20, 10);
+
+calc.multiply = function(x, y){
+  console.log(x * y);
+}
+calc.multiply(7, 7);
+
+calc.divide = function(x, y){
+  console.log(x / y)
+}
+calc.divide(25, 5);
+
+//Q9 返り値
+function remainder(x, y){
+  return x % y;
+}
+let result = remainder (5, 3);
+console.log('5 を 3 で割った余りは ' + result + ' です。');
+
+//Q10 スコープ
+//スコープというのは、変数の有効範囲のことで、関数の中で定義した変数は関数の中からしか参照できないようになっている。今回の場合はconsole.log(x);が関数の外でコンソールされているため、定義されていないと認識されエラーが出ている状況となっております。また、出力させるためには、頭の部分にfoo()というふうに関数の定義をしてあげる必要がある。
+
+//応用編 問題
+
+//Q1 標準組み込みオブジェクト
+let random = Math.floor(Math.random() * 10);
+console.log(random)
+
+//Q2 コールバック関数
+setTimeout(function(){
+  console.log('Hello World!');
+},3000);
+
+//Q3 if
+let num = 26;
+if(num >0){
+  console.log('num is greater than 0');
+} else if (num < 0){
+  console.log('num is less than 0');
+} else if (num === 0){
+  console.log('num is 0');
+}
+
+//Q4 for
+let numbers = [];
+
+for (let i = 0; i < 100; i++){
+  numbers.push(i);
+}
+console.log(numbers);
+
+//Q5 for × if
+let mixed = [4, '2', 5, '8', '9', 0, 1];
+for (let i = 0; i < mixed.length; i++){
+  let value = mixed[i];
+
+  if(typeof value === 'number'){
+    if(value % 2 === 0){
+    console.log('even');
+  } else{
+    console.log('odd');
+  } 
+} else{
+  console.log('not number');
+}
```

Comment:
下記のインデント崩れで、どのelseがどのifを指しているか不明瞭になっていますね。
基本的にifとelseのインデントは同一ラインで揃えましょう。
```suggestion
  if(typeof value === 'number'){
    if(value % 2 === 0){
      console.log('even');
    } else{
      console.log('odd');
    } 
  } else{
    console.log('not number');
  }

Reply:
インデントを修正しました。